### DISCUSSION TITLE
DataGrid Save Function close and run another Method for saving

### DISCUSSION THREAD

**Discussion by peterhym21** [Megabit/Blazorise#3768] [Q&A] [answered]
How do you get the Save button in the Inline Editor to close the editor and call another method in the @code{} block
and also how do you bind all the changes to a object,
Please help

                            <DataGrid TItem="UiOperationListVm" 
                            Data="@Operations" 
                            @bind-SelectedRow="@SelectedOperation" 
                            EditMode="DataGridEditMode.Inline" 
                            Editable Responsive>
                                <DataGridColumn Field="@nameof(UiOperationListVm.Id)" Caption="ID" Sortable="false" />
                                <DataGridColumn Field="@nameof(UiOperationListVm.Name)" Caption="Name" Editable />
                                <DataGridColumn Field="@nameof(UiOperationListVm.Description)" Caption="Description" Editable />
                                <DataGridCommandColumn>
                                    <EditCommandTemplate>
                                        <Button Color="Color.Primary" Clicked="@context.Clicked">Edit</Button>
                                    </EditCommandTemplate>
                                    <SaveCommandTemplate>
                                        <Button Color="Color.Primary" id="opbut" Clicked="@context.Clicked">Save</Button>
                                        // Run Method in code{} to update for a mediator
                                    </SaveCommandTemplate>

                                </DataGridCommandColumn>
                            </DataGrid>

**Comment by stsrki** [comment, marked as answer]
Something like this

```cshtml
<EditCommandTemplate>
    <Button Color="Color.Primary" Clicked="@(()=>OnEditClicked(context))">@context.LocalizationString</Button>
</EditCommandTemplate>
@code {
    async Task OnEditClicked( EditCommandContext<Employee> context )
    {
        // do something here
        await context.Clicked.InvokeAsync();
    }
}
```
